go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -301,62 +312,44 @@ func (npw *nodePortWatcher) updateServiceFlowCache(service *corev1.Service, netI
 				// set to Local, and the backend pod is HostNetworked. We need to add
 				// Flows that will DNAT all traffic coming into nodeport to the nodeIP:Port and
 				// ensure that the return traffic is UnDNATed to correct the nodeIP:Nodeport
-				if isServiceTypeETPLocal && hasLocalHostNetworkEp {
+				if isServiceTypeETPLocal && !localHostNetworkEndpoints.Empty() {
```

Comment:
note: above line 296 lacks coverage for unit tests:
flowProtocols = append(flowProtocols, protocol+"6")  <--- this

Reply:
also, the part below isUDNEnabledService completely lacks unit tests